crates/node/src/manager.rs

```diff
@@ -101,27 +113,157 @@ where
     /// sure any lefteover tasks are ended.  This allows it to be called more
     /// than once per program execution to support changing modes of the
     /// running node.
-    pub fn run(&mut self) -> eyre::Result<()> {
+    pub async fn run(&mut self) -> eyre::Result<()> {
         // create dbs to survive between sync state transitions
         let reth_db =
             RethEnv::new_database(&self.builder.node_config, self.tn_datadir.reth_db_path())?;
-        let consensus_db = open_db(self.tn_datadir.consensus_db_path());
 
-        // start initial epoch
+        // open the consensus db for this epoch
+        //
+        // TODO: consensus db should be epoch-specific - see issue #269
+        let consensus_db_path = self.tn_datadir.consensus_db_path();
+        info!(target: "epoch-manager", ?consensus_db_path, "node storage opened for epoch");
+        let consensus_db = open_db(consensus_db_path);
+
+        // Main task manager that manages tasks across epochs.
+        // Long-running tasks for the lifetime of the node.
+        let mut node_task_manager = TaskManager::new(NODE_TASK_MANAGER);
+        let node_task_spawner = node_task_manager.get_spawner();
+        // create oneshot channel to await the first epoch to start
+        let (start, ready) = oneshot::channel();
+        node_task_spawner.spawn_task("epoch-started", ready);
+
+        // track the status for when the node is ready
+        self.node_ready = Some(start);
+        // ensure task is tracked
+        node_task_manager.update_tasks();
```

Comment:
This should maybe go after spawn_node_networks() but since you join right after that anyway I don't think it matters and could maybe go away...  This mostly makes sure the proper tasks print when logging before the join, so not a functional issue and we are not logging before the join anyway.